pkg/datapath/loader/loader_test.go

```diff
@@ -35,12 +36,48 @@ type LoaderTestSuite struct{}
 var (
 	_              = Suite(&LoaderTestSuite{})
 	contextTimeout = 5 * time.Minute
+
+	dirInfo *directoryInfo
+	ep      = &testEP{}
 )
 
 func Test(t *testing.T) {
 	TestingT(t)
 }
 
+// runTests configures devices for running the whole testsuite, and runs the
+// tests. It is kept separate from TestMain() so that this function can defer
+// cleanups and pass the exit code of the test run to the caller which can run
+// os.Exit() with the result.
+func runTests(m *testing.M) int {
+	testIncludes = "-I/usr/include/x86_64-linux-gnu/"
+
+	tmpDir, err := ioutil.TempDir("/tmp/", "cilium_")
+	if err != nil {
+		log.Fatalf("Failed to create temporary directory: %s", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	if dirInfo, err = getDirs(tmpDir); err != nil {
+		log.Fatal(err.Error())
```

Comment:
Convert to return (int, error) and put the `log.Fatal()` in the caller to ensure that the deferred cleanup runs.